File: frontend/src/types/recommendations.ts
import type { Insight } from "./insights";

export interface Recommendation extends Insight {
  recommendation_type: string;
  subject_type: string;
  subject_id: string | null;
  action: string;
  title: string;
  reason: string;
}

export interface RecommendationListResponse {
  generated_at: string;
  workspace_id: string | null;
  recommendations: Recommendation[];
  read_only: boolean;
}

File: backend/app/schemas/recommendations.py
import uuid

from pydantic import BaseModel, Field

from app.schemas.insights import InsightResponse


class RecommendationResponse(InsightResponse):
    recommendation_type: str
    subject_type: str
    subject_id: uuid.UUID | None = None
    action: str
    title: str
    reason: str


class RecommendationListResponse(BaseModel):
    generated_at: str
    workspace_id: uuid.UUID | None = None
    recommendations: list[RecommendationResponse] = Field(default_factory=list)
    read_only: bool = True
